### PR TITLE
Adjust Primitives

### PR DESCRIPTION
**Pull request by AurevoirXavier**
I didn't see too much usage of dp-fee.
And it should a part of fee-market.
Like `staking::Ledger`, `democracy::Vote` which were defined in the pallet.

Also, we can easily integrate it into parachain through a single pallet.

### CODE REVIEW

**Comment by AsceticBear** [comment]
RED CI

File: frame/fee-market/src/lib.rs [hunk 1]
```diff
@@ -246,13 +245,14 @@ pub mod pallet {
 
 			// Increase the locked collateral
 			if new_collateral >= Self::relayer(&who).collateral {
-				let _ = T::RingCurrency::extend_lock(
+				T::RingCurrency::set_lock(
```
Comment (AsceticBear) [inline frame/fee-market/src/lib.rs:248]: I need some time to check the change here, please fix ci firstly.

**Review by HackFisher — APPROVED** [review]
LGTM

File: frame/fee-market/src/lib.rs (hunk 1 above)
Comment (AurevoirXavier) [reply frame/fee-market/src/lib.rs:248]: If you have already compared the value. Just set a new lock here and you can get rid of this error handling.

Comment (AsceticBear) [reply frame/fee-market/src/lib.rs:248]: I see the concrete implementation difference in this.  LGTM, just waiting for the ci.

**Review by AsceticBear — APPROVED** [review]

**Pull request merged**